uuid-cluster-dns/README.md

```diff
@@ -0,0 +1,60 @@
+# Using UUIDs for cluster endpoint domains
+
+The current approach for building cluster domains is to include the workload cluster's ID as a subdomain of the management cluster it belongs to, e.g.:
+
+```
+https://api.j9yr2.k8s.gauss.eu-west-1.aws.gigantic.io
+```
+
+This approach has a few drawbacks:
+
+1. The workload cluster is tied to the management cluster, making migration messy
+2. Workload clusters cannot be "promoted" to management clusters as the domain would no longer make sense
+3. Several customers have their own domain set up, leading to inconsistency between MCs (e.g. `eu-central-1.aws.cps.customer.com` vs. `gauss.eu-west-1.aws.gigantic.io`)
+4. The provider (and region) is baked into the domain, effectively preventing multi-region (or multi-cloud)
+
+An alternative "flat" DNS structure has been proposed that removes the link between WC and MC. E.g.:
+
+```
+https://api.j9yr2.k8s.[customer name].gigantic.io
+```
+
+While this solves some of the above drawbacks (1, 2, 4) it introduces its own complications:
+
+1. The customer name (or codename) is exposed in the domain
+2. There is no enforced global uniqueness on cluster IDs so if a customer has multiple MCs there is a chance that two workload clusters have the same ID, leading to conflicting domains
+
+## Proposal
+
+> Assumption: this is relevant only going forward with CAPI clusters and not to be "backported" to the vintage product.
+
+The flat DNS structure can be improved to avoid the drawbacks above my leveraging a UUID for the cluster domain and dissasociate the cluster ID/Name from URLs.
+
+E.g.
+
+```
+https://api.4c6aba87-061c-458d-91b6-62ad07979728.k8s.gigantic.io
+```
+
+For this to work, we'd need to generate a new UUID for a cluster either when running `kubectl-gs template cluster` (for use cases where GitOps are used) or via some defaulting logic when clusters CRs are applied.
```

Comment:
`kubectl-gs template cluster` will generate an AppCR and the configmap with this values.yaml, correct? 
https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/values.yaml

Cluster CRs won't get applied against the API directly anymore.

Ok, I've found it here for openstack: https://github.com/giantswarm/kubectl-gs/blob/891b222d7f003731005f09d95e86e236cabfa1f5/cmd/template/cluster/provider/capo.go#L28

@marians does it make sense to generate the unique identifier in the values yaml? I guess it needs to be this way because the AppCR AppName needs to be unique too and at least at the moment is generated by the cluster name within `kubectl gs` already.

https://github.com/giantswarm/kubectl-gs/blob/891b222d7f003731005f09d95e86e236cabfa1f5/cmd/template/cluster/provider/capo.go#L117